R/AnnotationGenome.R

```diff
@@ -24,23 +24,30 @@ createGenomeAnnotation <- function(
   .validInput(input = filter, name = "filter", valid = c("boolean"))
   .validInput(input = filterChr, name = "filterChr", valid = c("character", "null"))
 
-  if(is.null(genome) | is.null(blacklist) | is.null(chromSizes)){
-
-    ##################
-    message("Getting genome..")
-    bsg <- validBSgenome(genome)
-    genome <- bsg@pkgname
-
-    ##################
-    message("Getting chromSizes..")
+  ##################
+  message("Getting genome..")
+  #validBSgenome works on both character and BSgenome inputs, which are the only allowable inputs to the param
+  bsg <- validBSgenome(genome)
+  genome <- bsg@pkgname
+
+  if(is.null(chromSizes)) {
+    message("Attempting to infer chromSizes..")
     chromSizes <- GRanges(names(seqlengths(bsg)), IRanges(1, seqlengths(bsg)))
     if(filter){
-        chromSizes <- filterChrGR(chromSizes, remove = filterChr)
+      if(is.null(filterChr)) {
```

Comment:
This isnt necessary because filterChrGR can handle null input

Reply:
i removed the if statement since it would prevent removal of non-standard chromsomes without manually specifying the list of chromsomes